src/apple/macos/ffi.rs

```diff
@@ -38,89 +43,57 @@ extern "C" {
 
     pub fn IOObjectRelease(obj: io_object_t) -> i32;
 
-    pub fn IOServiceOpen(device: io_object_t, a: u32, t: u32, x: *mut io_connect_t) -> i32;
-
-    pub fn IOServiceClose(a: io_connect_t) -> i32;
-
-    #[allow(dead_code)]
-    pub fn IOConnectCallStructMethod(
-        connection: mach_port_t,
-        selector: u32,
-        inputStruct: *const KeyData_t,
-        inputStructCnt: size_t,
-        outputStruct: *mut KeyData_t,
-        outputStructCnt: *mut size_t,
-    ) -> i32;
-    // pub fn IORegistryEntryCreateCFProperties(
-    //     entry: io_registry_entry_t,
-    //     properties: *mut CFMutableDictionaryRef,
-    //     allocator: CFAllocatorRef,
-    //     options: IOOptionBits,
-    // ) -> kern_return_t;
-    // pub fn IORegistryEntryGetName(entry: io_registry_entry_t, name: *mut c_char) -> kern_return_t;
-}
-
-extern "C" {
-    pub fn CFStringCreateWithCStringNoCopy(
-        alloc: *mut c_void,
-        cStr: *const c_char,
-        encoding: CFStringEncoding,
-        contentsDeallocator: *mut c_void,
-    ) -> CFStringRef;
-
-    // Disk information functions are non-operational on iOS because of the sandboxing
-    // restrictions of apps, so they don't can't filesystem information. This results in
-    // mountedVolumeURLs and similar returning `nil`. Hence, they are MacOS specific here.
-
-    pub fn DASessionCreate(allocator: CFAllocatorRef) -> DASessionRef;
-
-    // pub fn DADiskCreateFromVolumePath(
-    //     allocator: CFAllocatorRef,
-    //     session: DASessionRef,
-    //     path: CFURLRef,
-    // ) -> DADiskRef;
-    pub fn DADiskCreateFromBSDName(
+    pub fn IORegistryEntryCreateCFProperty(
+        entry: io_registry_entry_t,
+        key: CFStringRef,
         allocator: CFAllocatorRef,
-        session: DASessionRef,
-        path: *const c_char,
-    ) -> DADiskRef;
-    // pub fn DADiskGetBSDName(disk: DADiskRef) -> *const c_char;
-
-    pub fn DADiskCopyDescription(disk: DADiskRef) -> CFMutableDictionaryRef;
-}
-
-pub type DADiskRef = *const __DADisk;
-pub type DASessionRef = *const __DASession;
-
-// We need to wrap `DASessionRef` to be sure `System` remains Send+Sync.
-pub struct SessionWrap(pub DASessionRef);
-
-impl Drop for SessionWrap {
-    fn drop(&mut self) {
-        if !self.0.is_null() {
-            unsafe {
-                CFRelease(self.0 as _);
-            }
-        }
-    }
+        options: IOOptionBits,
+    ) -> CFDictionaryRef;
+    pub fn IORegistryEntryGetParentEntry(
+        entry: io_registry_entry_t,
+        plane: io_name_t,
+        parent: *mut io_registry_entry_t,
+    ) -> libc::kern_return_t;
+
+    pub fn IOBSDNameMatching(
+        mainPort: mach_port_t,
+        options: u32,
+        bsdName: *const c_char,
```

Comment:
```suggestion
        bsdName: *const cchar,
```

Again, `cchar` isn't the same as `char`.

Reply:
Resolving per the conversation about about Swift vs Objective-C types.